analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,357 @@
+  filter(cytoband %in% neutral_cytoband_status_df$cytoband)
+
+if (!(running_in_ci)) {
+  # Now count the gene level calls (for each status call) and define
+  # the gene as that status if more than 50% of the total counts are
+  # for that particular status
+  gene_status_count <- status_majority_caller(
+    consensus_seg_autosomes_df_develop,
+    "gene_symbol"
+  )
+
+  # Display table
+  gene_status_count
+}
+```
+
+## Plot calls
+
+Plot the final dominant status call on the x-axis and the percent of each status on the y-axis.
+
+### Plot chromosome arm status calls
+
+```{r}
+# Run `plot_dominant_status_calls` function for each specific status' set of
+# percentage values
+arm_gain_status_plot <- plot_dominant_status_calls(
+  arm_status_count,
+  "percent_gain"
+)
+arm_gain_status_plot
+
+arm_loss_status_plot <- plot_dominant_status_calls(
+  arm_status_count,
+  "percent_loss"
+)
+arm_loss_status_plot
+
+arm_neutral_status_plot <- plot_dominant_status_calls(
+  arm_status_count,
+  "percent_neutral"
+)
+arm_neutral_status_plot
+```
+
+### Plot cytoband status calls
+
+```{r}
+# Run `plot_dominant_status_calls` function for each specific status' set of
+# percentage values
+cytoband_gain_status_plot <- plot_dominant_status_calls(
+  cytoband_status_count,
+  "percent_gain"
+)
+cytoband_gain_status_plot
+
+cytoband_loss_status_plot <- plot_dominant_status_calls(
+  cytoband_status_count,
+  "percent_loss"
+)
+cytoband_loss_status_plot
+
+cytoband_amplification_status_plot <- plot_dominant_status_calls(
+  cytoband_status_count,
+  "percent_amplification"
+)
+cytoband_amplification_status_plot
+```
+
+### Plot gene status calls
+
+```{r}
+# Plot if not running in circleCI
+if (!(running_in_ci)) {
+  gene_gain_status_plot <- plot_dominant_status_calls(gene_status_count,
+                                                      "percent_gain")
+  
+  gene_loss_status_plot <- plot_dominant_status_calls(gene_status_count,
+                                                      "percent_loss")
+  
+  gene_amplification_status_plot <- plot_dominant_status_calls(gene_status_count,
+                                                               "percent_amplification")
+  
+  # Display plots using the `list` function
+  list(gene_gain_status_plot,
+       gene_loss_status_plot,
+       gene_amplification_status_plot)
+}
+```
+
+## Combine arm, cytoband, and gene level status data
+
+```{r}
+# The logic variable `running_in_ci` is needed here because the CI testing
+# files do not contain any of the genes in the `gene_status_count` data.frame
+# generated above (when `running_in_ci` == FALSE)
+if (!(running_in_ci)) {
+  final_df <- consensus_seg_autosomes_df %>%
+    mutate(chromosome_arm = gsub("(p|q).*", "\\1", cytoband)) %>%
+    left_join(arm_status_count, by = "chromosome_arm") %>%
+    left_join(cytoband_status_count, by = "cytoband") %>%
+    left_join(gene_status_count, by = "gene_symbol")
```

Comment:
Can you use a `suffix` argument here and have them labeled with their origins? right now it looks like they have the default `.x` and `.y`s.

Reply:
We may also want to have a final column that summarizes which status we should be listening to. Aka if `arm_status` is declared, report that, then `cytoband`, then `gene_symbol`. But, we may want some more information about the downstream applications for this for knowing on how this final table should be formatted.